src/common/dump.c

```diff
@@ -20,24 +20,23 @@ static inline void dump_regs_SCB(void) {
 }
 
 void dump_to_xflash(void) {
-    _Static_assert(sizeof(dumpinfo_t) == 16, "invalid sizeof(dumpinfo_t)");
-    uint32_t addr;
-    if (!dump_in_xflash_is_empty()) {
-        if (dump_in_xflash_is_saved()) {
-            if (!dump_in_xflash_is_displayed()) {
-                return;
+    if (w25x_init(false)) {
```

Comment:
Can't we just early-exit,  instead of adding one more indentation level to the whole function?

Reply:
Sure we can.

Reply:
fixed